service/health/searcher/azure.go

```diff
@@ -2,12 +2,33 @@ package searcher
 
 import (
 	"context"
+
+	"github.com/giantswarm/microerror"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // searchAzureCR searches for the cluster config in AWSClusterConfigs resources.
 func (s *Service) searchAzureCR(ctx context.Context, request Request) (*Response, error) {
+	azureCR, err := s.g8sClient.
+		ProviderV1alpha1().
+		AzureConfigs("default").
+		Get(request.ClusterID, v1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	var health string
+	{
+		nodeCount := len(azureCR.Status.Cluster.Nodes)
```

Comment:
Fine as the very first implementation. We should start thinking about how we split this logic. 
The logic for health calculation is pretty much the same for all providers - so how can we reuse this?

Reply:
This is what I'm working on now. Will discuss with them team once I have some more concrete ideas.